packages/core/src/messagingProtocols/mls/E2EIdentityService/E2EIServiceExternal.ts

```diff
@@ -78,21 +68,49 @@ export class E2EIServiceExternal {
     return this.coreCryptoClient.e2eiIsEnabled(ciphersuite);
   }
 
-  public async getUsersIdentities(groupId: string, userIds: QualifiedId[]) {
-    return this.coreCryptoClient.getUserIdentities(
+  public async getUsersIdentities(groupId: string, userIds: QualifiedId[]): Promise<Map<string, DeviceIdentity[]>> {
+    // We keep track of the ID we give to CoreCrypto in order to map it back to regular userIds afterwards
+    const hashedIdMap = new Map(userIds.map(userId => [uuidTobase64url(userId.id).asString, userId]));
+
+    const userIdentities = await this.coreCryptoClient.getUserIdentities(
       Decoder.fromBase64(groupId).asBytes,
-      userIds.map(userId => uuidTobase64url(userId.id).asString),
+      Array.from(hashedIdMap.keys()),
     );
+
+    const mappedUserIdentities = new Map();
+    for (const [base64Id, identities] of userIdentities) {
+      // remapping coreCrypto user ids to regular userIds
+      const userId = hashedIdMap.get(base64Id);
+      if (userId) {
+        mappedUserIdentities.set(
+          userId.id,
+          identities.map(identity => ({
+            ...identity,
+            deviceId: parseFullQualifiedClientId((identity as any).client_id).client,
```

Comment:
The types from CoreCrypto are wrong. The interface says the property is `clientId` but the payload actually contains `client_id` 🙃